File: java/version_00/counter/Counter.java
package counter;

import java.util.Scanner;

public class Counter{
	
	public Counter(){ counter = 0; }

	/**********************
			Getters
	**********************/

	public int getCounter(){ return this.counter; }

	/**********************
			Setters
	**********************/

	public void setCounter(int c){ this.counter = c; }

	public void incrementCounter(){ this.counter++; }
	public void decrementCounter(){ this.counter--; }

	private int counter;
}

File: c/version_00/counter/Counter.c
#include <stdio.h>
#include "Counter.h"

/**********************
		Getter(s)
**********************/

int getCounter(Counter c){
	return c.counter;
}

/**********************
		Setter(s)
**********************/

void setCounter(Counter *c, int counter){
	c->counter = counter;
}
void incrementCounter(Counter *c){
	c->counter++;
}
void decrementCounter(Counter *c){
	c->counter--;
}